apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsCard/index.tsx

```diff
@@ -130,14 +158,37 @@ const CommitDetailsCard: React.FC<CommitDetailsCardProps> = ({
         >
           <Icon glyph="GitHub" />
         </IconButton>
-        <Button
-          data-cy="restart-button"
-          disabled={!canRestart}
-          onClick={() => restartTask()}
-          size={ButtonSize.XSmall}
-        >
-          Restart Task
-        </Button>
+        {activated ? (
+          <Button
+            data-cy="restart-button"
+            disabled={!canRestart}
+            onClick={() => {
+              restartTask();
+              sendEvent({
+                name: "Clicked restart task button",
+                "task.id": task.id,
```

Comment:
This value may be too unique to actually be useful in general analytics workflows. What if we logged the task name/variant? 

That way we can query for most restarted tasks or variants via the task history page

Reply:
ok i will add both!